### PR TITLE
[WIP] Fix docker push issues

### PR DESCRIPTION
**Pull request by rbren**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Forked PRs don't have permission to push to our GHCR repo. 

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This changes the workflow in those cases to only build the image.

---
**Other references**

### CODE REVIEW

**Review by tobitege — APPROVED** [review]
LGTM, thanks!

**Comment by tobitege** [comment]
Once this works, maybe we can check again to not hard-code repo names so it works on forks still?

File: .github/workflows/ghcr_runtime.yml [hunk 1]
```diff
@@ -70,12 +76,27 @@ jobs:
...
         run: |
-          suffix=$(echo "${{ matrix.base_image }}" | cut -d ':' -f 1 | cut -d '/' -f 1)
-          ./containers/build.sh runtime ${{ github.repository_owner }} --push $suffix
+          ./containers/build.sh runtime ${{ github.repository_owner }} --push ${{ matrix.base_image.tag }}
+
+      # Forked repos can't push to GHCR, so we need to upload the image as an artifact
+      - name: Build runtime image ${{ matrix.base_image.image }} for fork
+        uses: docker/build-push-action@v6
+        with:
+          tags: ghcr.io/all-hands-ai/runtime:${{ github.sha }}-${{ matrix.base_image.tag }}
```
Comment (tobitege) [inline .github/workflows/ghcr_runtime.yml:90]: replace `all-hands-ai` with `${{ toLowerCase(github.repository_owner) }}` ?

**Review by tobitege — APPROVED** [review]
LGTM, just nit about repo name

File: .github/workflows/ghcr_runtime.yml (hunk 1 above)
Comment (tobitege) [reply .github/workflows/ghcr_runtime.yml:90]: Oh, forgot to mention, that the `toLowerCase` is supposed to be an action. But yeh, if it poses more issues, than of course skip that. :)

**Comment by tobitege** [comment]
Fyi, restarted some tests, took already over an hour.
Also this:
grafik

File: .github/workflows/ghcr_runtime.yml [hunk 2]
```diff
@@ -96,6 +117,19 @@ jobs:
           haskell: true
           large-packages: true
           swap-storage: true
+
+      # Forked repos can't push to GHCR, so we need to download the image as an artifact
+      - name: Download runtime image for fork
```
Comment (xingyaoww) [inline .github/workflows/ghcr_runtime.yml:122]: You probably also need to add this Job for run integration tests?

Comment (rbren) [reply .github/workflows/ghcr_runtime.yml:122]: good catch!

**Pull request merged**